core/src/main/java/hudson/Functions.java

```diff
@@ -201,7 +202,25 @@ public static String xsDate(Calendar cal) {
     public static String rfc822Date(Calendar cal) {
         return Util.RFC822_DATETIME_FORMATTER.format(cal.getTime());
     }
-    
+
+    /**
+     * During Jenkins start-up, before {@link InitMilestone#PLUGINS_STARTED} the extensions lists will be empty
+     * and they are not guaranteed to be fully populated until after {@link InitMilestone#EXTENSIONS_AUGMENTED}.
+     * If you attempt to access the extensions list from a UI thread while the extensions are being loaded you will
+     * hit a big honking great monitor lock that will block until the effective extension list has been determined
+     * (as if a plugin fails to start, all of the failed plugin's extensions and any dependent plugins' extensions
+     * will have to be evicted from the list of extensions. In practical terms this only affects the
+     * "Jenkins is loading" screen, but as that screen uses the generic layouts we provide this utility method
```

Comment:
My test system would go nuts if the web load got blocked at this point as you would have a big pile of request threads all bunching up because they all hit the lock at the same time and thus were released at the same time. Fixing this enabled me to get consistent failure from the test system under randomised UI load. I could take it out of the pull request, but that would have complicated my verification testing that the failure modes were fixed by the overall pull request